conf/config.go

```diff
@@ -36,8 +36,41 @@ var (
 
 	// Vendor services url, used by osie to proxy requests for OS image artifacts
 	OsieVendorServicesURL = env.Get("OSIE_VENDOR_SERVICES_URL")
+
+	// BOOTS_IPXE_VARS can optionally be used to inject any number of variables
+	// into the iPXE script
+	DynamicIPXEVars = parseDynamicIPXEVars(env.Get("BOOTS_IPXE_VARS"))
```

Comment:
We're trying to move away from the pattern this file follows of calling environment variables. 

_"Only func main has the right to decide which flags, env variables, config files are available to the user "_ -- [ref](https://github.com/tinkerbell/boots/blob/main/docs/DESIGNPHILOSOPHY.md)

Boots' design philosophy points env var to func main. Ideally, i think we would want to have this be a CLI flag/env combo and then plumbed through to these installers. See this [func](https://github.com/tinkerbell/boots/blob/51fec55b27aa94deda3367e717bb19016d7eec0c/cmd/boots/main.go#L366) for the others using this pattern.

Reply:
Hey Jacob! We have an interesting issue where we're downloading boot assets via an https url, but the host making the request may have a networking configuration restricted to Layer 2, and thus be unable to validate the tls cert from ca.ixpe.org. We do have a local proxy service available, but to use it we need to set the iPXE var `crosscert`.

Originally the idea was to add yet another env var to boots, which we know is not optimal. But with this approach, at least we can add only one env var and potentially solve this class of problem again without touching more boots code. If `BOOTS_IPXE_VARS` isn't set, then the ipxe scripts will remain the same as before. But now we can do things in the helm chart such as:

```
env:
  - name: BOOTS_IPXE_VARS
    value: |
      vendors_domain=vendors.{{ .clusterInfo.fqdn }}
      crosscert=http://${vendors_domain}/ipxe/ca
```

That's the background on the motivation behind this. Let me see what I can do to make this a CLI flag according to your suggestion.

Reply:
right on, thank you for sharing that! very helpful. I had forgotten that `set` in ipxe does more than just arbitrary variable setting.

Would you also be open to adding a code comment with a link to the ipxe docs around variables like this? https://ipxe.org/cfg